### PR TITLE
[BUG]:The Set Due Date dialog box is missing its title 

### LINKED ISSUE #17124: [BUG]: The "Set Due Date" dialog box is missing its title
### Checked for duplicates?

- [X] This issue is not a duplicate

### Does it also happen in the desktop version?

- [X] This bug does not occur in the latest version of Anki Desktop

### What are the steps to reproduce this bug?

settings-> controls->set due date 

screenshot2

settings-> appearance->app bar button-> set due date
WhatsApp Image 2024-09-26 at 4 40 43 PM

### Expected behaviour

add missing  title

### Debug info

```text
debug
```

### (Optional) Anything else you want to share?

_No response_

### Research

- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by gautamsingh2**
## Purpose / Description

The Set Due Date dialog box is missing its title .
## Fixes
* Fixes #17124 

##Screenshot
settings-> controls->set due date
WhatsApp Image 2024-09-26 at 4 27 46 PM

settings-> appearance->app bar button-> set due date
WhatsApp Image 2024-09-26 at 4 27 45 PM

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by criticalAY** [comment]
This PR is irrelevant since these a tool string as are visible in dev build only, actual strings are coming from backend only

**Review by david-allison — CHANGES_REQUESTED** [review]
This should come from the backend.

* update the xml files to note that this is the reason the strings are hardcoded
* update the code to set the title using (I believe): `TR.actionsSetDueDate().toSentenceCase(this@SetDueDateDialog, R.string.sentence_set_due_date))`

**Comment by gautamsingh2** [comment]
review feedback done

**Review by david-allison — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/ControlsSettingsFragment.kt [hunk 1]
```diff
@@ -45,7 +45,11 @@ class ControlsSettingsFragment : SettingsFragment() {
 
     private fun setDynamicTitle() {
         findPreference<ControlPreference>(getString(R.string.reschedule_command_key))?.let {
-            it.title = TR.actionsSetDueDate().toSentenceCase(R.string.sentence_set_due_date)
+            it.apply {
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/preferences/ControlsSettingsFragment.kt:48]: There's no need for this, just use it.title and it.dialogTitle.

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/CustomButtonsSettingsFragment.kt [hunk 2]
```diff
@@ -59,8 +60,12 @@ class CustomButtonsSettingsFragment : SettingsFragment() {
     }
 
     private fun setDynamicTitle() {
-        findPreference<Preference>(getString(R.string.custom_button_schedule_card_key))?.let {
-            it.title = TR.actionsSetDueDate().toSentenceCase(this, R.string.sentence_set_due_date)
+        findPreference<ListPreference>(getString(R.string.custom_button_schedule_card_key))?.let {
+            val preferenceTitle = TR.actionsSetDueDate().toSentenceCase(this, R.string.sentence_set_due_date)
+            it.apply {
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/preferences/CustomButtonsSettingsFragment.kt:65]: Same, just use it.*

**Review by lukstbit — CHANGES_REQUESTED** [review]

**Review by lukstbit — APPROVED** [review]
Thanks!

**Comment by Aminadev2** [comment]
Hi @david-allison ,i also get deeply disappointed . other paid for their work .But not me.

**Comment by Aminadev2** [comment]
Good Bye

**Pull request merged**